### ISSUE TITLE
Http Digest Authentication Support

### ISSUE THREAD

**Issue by dwen77** [springfox/springfox#2471]
Hi,

I am working on an API and it requires http digest authentication, however, it is not supported in the latest version of springfox (version 2.9.0). Therefore, the Swagger UI cannot show the proper authentication method. My question is that if there is some other ways to support digest authentication or it is on the roadmap?

Thanks in advance!

**Comment by dilipkrish** [comment]
Digest authentication for the API or for protecting swagger-ui?

**Comment by dwen77** [comment]
Hi Dilipkrish,

Thanks for the answer! I meant the digest authentication for the APIs.

**Comment by dilipkrish** [comment]
Just to clarify, you should be using spring security to secure your API. And describe that for the specification by configuring `BasicAuth` security using springfox. Is that what you're doing?

**Comment by dwen77** [comment]
Yes, it is what I am doing now. Could it give some confusions if we describe it by configuring `BasicAuth` but the actual authentication is `DigestAuth?` Moreover, it asks for credential when clicking 'try it out' even though the `BasicAuth` has been configured with valid credential in Swagger UI already.

**Comment by dilipkrish** [comment]
Yes possibly an issue with [Swagger-ui](https://github.com/swagger-api/swagger-ui)

**Comment by dwen77** [comment]
Yes, I agree. I consider it as a limitation of Swagger-ui. I would prefer to hide the authentication from Swagger UI and describer it elsewhere to avoid misleading.
